src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -208,6 +209,16 @@ public Snapshot retrieveAvailableSnapshot(UUID snapshotId) {
         return retrieveSnapshot(snapshotId, true);
     }
 
+    /**
+     * This is a convenience wrapper that returns a snapshot project model only if it is NOT exclusively locked.
+     * This method is intended for user-facing API calls (e.g. from RepositoryApiController).
+     * @param snapshotId the snapshot id
+     * @return the SnapshotProjectModel object
+     */
+    public SnapshotProjectModel retrieveAvailableSnapshotProject(UUID snapshotId) {
```

Comment:
This should have the `@Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE, readOnly = true)` annotation (as should all other public get methods in this class